modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -694,14 +696,19 @@ void generateApis(List<File> files, List<OperationsMap> allOperations, List<Mode
                 addAuthenticationSwitches(operation);
 
                 for (String templateName : config.apiTemplateFiles().keySet()) {
-                    File written;
+                    File written = null;
                     if (config.templateOutputDirs().containsKey(templateName)) {
                         String outputDir = config.getOutputDir() + File.separator + config.templateOutputDirs().get(templateName);
                         String filename = config.apiFilename(templateName, tag, outputDir);
-                        written = processTemplateToFile(operation, templateName, filename, generateApis, CodegenConstants.APIS, outputDir);
+                        // do not overwrite apiController file for spring server
+                        if (apiFilePreCheck(filename, generatorCheck, templateName, templateCheck)){
+                            written = processTemplateToFile(operation, templateName, filename, generateApis, CodegenConstants.APIS, outputDir);
+                        }
                     } else {
                         String filename = config.apiFilename(templateName, tag);
-                        written = processTemplateToFile(operation, templateName, filename, generateApis, CodegenConstants.APIS);
+                        if(apiFilePreCheck(filename, generatorCheck, templateName, templateCheck)){
+                            written = processTemplateToFile(operation, templateName, filename, generateApis, CodegenConstants.APIS);
+                        }
```

Comment:
can we add a `else` clause to log an info message saying the implementation file is not overwritten?

Reply:
yeah sure @wing328 , I will add and push the commit and Thanks for the review!

Reply:
I have pushed the required changes @wing328. Kindly check and merge.